### PR TITLE
Documentation for testing standards

### PR DESCRIPTION
**Pull request by adamszalapski**
## Jira

http://vjira2:8080/browse/BDS-1162

## Summary

Add documentation for testing standards.

## Details

Add two markdown files for JavaScript and manual testing standards.

## How to test

Check 'Testing Standards' section in 'DOCS'. Please give all your suggestions and concerns about this.
JS testing: https://feature-documentation-for-testing-standar.boltdesignsystem.com/docs/development/testing-standards/js-testing-standards.html
Manual testing: https://feature-documentation-for-testing-standar.boltdesignsystem.com/docs/development/testing-standards/manual-testing-standards.html

### CODE REVIEW

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+  const results = await render('@bolt-components-image/image.twig', {
+    src: '/fixtures/1200x660.jpg',
+    alt: 'A Rock Climber',
+    lazyload: true,
+  });
+  expect(results.ok).toBe(true);
+  expect(results.html).toMatchSnapshot();
+});
+```
+*Note: Fixtures folder must be present in the `__tests__` folder in specific component*
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: "`fixtures` folder must be present in the `__tests__` folder for the respective component"

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+    failureThreshold: '0.01',
+    failureThresholdType: 'percent',
+  });
+});
+```
+
+## Nightwatch test
+Nightwatch tests are not reusable and must be written for specific use case. Documentation how to use and create Nightwatch test is [here](https://nightwatchjs.org/guide).
+
+### How to check if component contain specific class
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: "contains a specific class"

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+      },
+    )
+    .saveScreenshot(
+      `screenshots/bolt-image/${testName}--${currentBrowser}.png`,
+    )
+    .end();
+}
+```
+
+### How to check if component exist, have specific class and contain some text
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: "if component exists, has specific class, and contains some text"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+---
+title: Manual testing standards
+---
+
+## Main Benefits
+
+- Test repeatability
+- Simple instructions that all testers can proceed
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "Intuitive instructions that all testers can follow"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+- The same language and wording used
+- Separate scenarios for tests
+
+## Best practises
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "practices"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+- create `TESTING.md` file in root of the component to add manual testing steps
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "in the component's root folder"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+---
+title: Manual testing standards
+---
+
+## Main Benefits
+
+- Test repeatability
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: Same comment as the other page: choose either **Sentence case** or **lowercase** for all bullet points.

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+- Simple instructions that all testers can proceed
+- The same language and wording used
+- Separate scenarios for tests
+
+## Best practises
+- create `TESTING.md` file in root of the component to add manual testing steps
+- use Gherkin format for creating steps in scenarios
+- always linking to https://boltdesignsuystem.com for reference in functionally and visually tests
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "always link to ... in functional and visual tests"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+## When create manual testing instructions
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "When to create"

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+- when tests need additional manual testing
```
Comment (mikemai2awesome) [inline docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: "when a component needs..."

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 11]
```diff
@@ -0,0 +1,504 @@
+---
+title: JavaScript testing standards
+---
+
+> The one imperfect test is better than no test at all.
+
+## Main Benefits
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md (hunk 7 above)
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 12]
```diff
@@ -0,0 +1,504 @@
...
+    - full: `yarn test:e2e:full-master`
+    - quick: `yarn test:e2e:quick-master`
+1. E2E specific test localy
+    - `NOW_URL='url-for-test' ./node_modules/.bin/nightwatch --config 'path-to-config' --env 'brwosers-list' --test 'path-to-e2e-test-file'` 
+    - example: `NOW_URL=http://localhost:3000 ./node_modules/.bin/nightwatch --config packages/testing/testing-nightwatch/nightwatch.local.js --env chrome,safari --test packages/components/bolt-accordion/__tests__/accordion.e2e.js`
+
+*Note: To run E2E local test you should watch bolt files by launching: `yarn start`.\
+Note 2: Monorepo test will fail when you earlier not compile bolt at least one time.*
+
+## JEST test
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 13]
```diff
@@ -0,0 +1,504 @@
...
+  const image = await page.screenshot();
+ 
+  expect(image).toMatchImageSnapshot({
+    failureThreshold: '0.01',
+    failureThresholdType: 'percent',
+  });
+});
+```
+
+## Nightwatch test
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md (hunk 1 above)
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md (hunk 2 above)
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md (hunk 3 above)
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md (hunk 4 above)
Comment (danielamorse) [reply docs-site/src/pages/docs/45-development/70-testing-standards/20-manual-testing-standards.md]: Done.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 14]
```diff
@@ -0,0 +1,504 @@
...
+test('Basic usage', async () => {
+  const results = await render('@bolt-components-button/button.twig', {
+    text: 'This is a button',
+  });
+  expect(results.ok).toBe(true);
+  expect(results.html).toMatchSnapshot();
+});
+```
+
+### How to create schema prop test
```
[the remark this answers is not in the document]
Comment (adamszalapski) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 15]
```diff
@@ -0,0 +1,504 @@
...
+```javascript
+const { tag } = schema.properties;
+ 
+tag.enum.forEach(async tagChoice => {
+  test(`Button tag: ${tagChoice}`, async () => {
```
[the remark this answers is not in the document]
Comment (adamszalapski) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done but maybe we should update this on real button test.

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md [hunk 16]
```diff
@@ -0,0 +1,504 @@
...
+    });
+    expect(results.ok).toBe(true);
+    expect(results.html).toMatchSnapshot();
+  });
+});
+```
+
+### How to create boolean prop test
+```javascript
+test('Button with "disabled" adds attr to <a>', async () => {
```
[the remark this answers is not in the document]
Comment (adamszalapski) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Done but we should update our test as well.

**Comment by adamszalapski** [comment]
Thanks @danielamorse i applied all @mikemai2awesome feedback that was left

File: docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: There's still a typo here with the `<a>`. 😄

[the remark this answers is not in the document]
Comment (adamszalapski) [reply docs-site/src/pages/docs/45-development/70-testing-standards/10-js-testing-standards.md]: Great eye @mikemai2awesome. Just fix it.

**Comment by sghoweri** [comment]
PR was released with v2.11.0

**Pull request merged**